content/blog/2022-05-10-compcert/index.md

```diff
@@ -0,0 +1,112 @@
+* **LTL** is similar to RTL except pseudo-registers are replaced by hardware registers or abstract stack locations after register allocation is performed. 
+* **LTLin** is the result of CFG linearization which produces a list of instructions with labels and branches.
+* **Linear** is the next IR after LTLin. Spill and reload instructions are placed around instructions that reference pseudo-registers allocated to stack locations. Move instructions are inserted to enforce calling conventions.
+* **Mach** is the result of laying out the activation records of functions.
+* **PPC** is similar to the PowerPC assembly language. Instruction scheduling is performed to go from Mach to PPC. Note that this pass uses the verified validator approach.
+
+### Proof of Correctness
+
+The source, intermediate, and target languages of CompCert all have formally defined semantics. The Coq proof assistant was used to prove that the transformations between these languages preserve semantics. The Coq formalization and proof amounts to 42,000 lines of Coq and approximately 3 person-years of work. 14% of these lines define compilation algorithms, 10% specify language semantics, and 76% correspond to the correctness proof.
+
+Coq's extraction facility was used to generate an executable compiler. This process involves translating functional specifications in Coq into Caml code and then running the Caml code through the Caml compiler. Note that the correctness of CompCert relies on trust in this compilation chain since it has not been formally verified.
+
+### Performance
+
+CompCert's performance turns out to be adequate for critical embedded code. CompCert was benchmarked against the GCC 4.0.1 compiler at optimization levels 0, 1, and 2. CompCert generates code that is twice as fast as code generated with GCC without optimization. CompCert code is 7% slower than gcc-01 and 12% slower than gcc-02 on average. Compilation times of CompCert are within a factor of 2 of those of gcc-01. Although CompCert's compilation times are higher than GCC's, and CompCert code runs slower than GCC code on average, this is a small price to pay for CompCert's correctness guarantees.
+
+<p align="center">
+<img src="fig2.png" alt="alt_text" title="image_tooltip" style="zoom:25%;" />
+</p>
+
+### The Story Since 2009
+
+CompCert has been received warmly by the community; the day after our presentation it won the 2021 ACM Software System Award. It remains in active development not only in the research community but also, in partnership with AbsInt, in a commercial setting. It now supports five target languages and not just one, and efforts have been made to extend the verification upwards to the parsing phase and downwards to the assembling phase.
```

Comment:
Maybe the Software System Award and AbsInt mentions could use links to where the reader can learn more?